### ISSUE TITLE
Portal 2 Bug: Fling Crash in sp_a3_end

### ISSUE THREAD

**Issue by JVGRYT** [StrataSource/Portal-2-Community-Edition#1678]
### Describe the bug

I ran into a bug during my gameplay of Portal 2 with only the paintgun (min portals). In the map `sp_a3_end`, there is a point further into the map where once you fling yourself, the game crashes. I've attached a video to make the location easier to explain.
https://github.com/StrataSource/Portal-2-Community-Edition/assets/92970602/f7700404-d6c8-4dc7-b6ab-bdc981cde67a
This crash makes the game unbeatable without using cheats.

### To Reproduce

1. Open `sp_a3_end`
2. Progress to the point in the map shown in the video
3. Use the gels to fling yourself
4. Crash

**Comment by pivotman319-owo** [comment]
crash confirmed on latest QA build, appears to only occur if weapon_paintgun is held by the player and if they granted themselves upgrades (via upgrade_paintgun) before switching out to a different weapon

**Comment by JVGRYT** [comment]
> crash confirmed on latest QA build, appears to only occur if weapon_paintgun is held by the player and if they granted themselves upgrades (via upgrade_paintgun) before switching out to a different weapon

That's quite intriguing, I'd be interested to know why that happens 🤔

**Comment by craftablescience** [comment]
This crash happens regardless of held weapons, however it is also random. You probably rolled the dice and got lucky the first time

**Comment by JVGRYT** [comment]
> This crash happens regardless of held weapons, however it is also random. You probably rolled the dice and got lucky the first time

Yes, but it is definitely consistent with the paintgun held, as far as I've tested it
That or I've been unlucky at least ten times in a row

However, I think I do recall this crash happening in a casual playthrough in P2:CE, but I reloaded the game and it worked fine so I ended up forgetting about it and I never made an issue about it

**Comment by craftablescience** [comment]
> > This crash happens regardless of held weapons, however it is also random. You probably rolled the dice and got lucky the first time
> 
> Yes, but it is definitely consistent with the paintgun held, as far as I've tested it That or I've been unlucky at least ten times in a row
> 
> However, I think I do recall this crash happening in a casual playthrough in P2:CE, but I reloaded the game and it worked fine so I ended up forgetting about it and I never made an issue about it

It has nothing to do with holding the paintgun, it's a crash in the physics code. I imagine it's more likely to crash than not. If you don't go through those portals you should be fine

**Comment by JVGRYT** [comment]
> > > This crash happens regardless of held weapons, however it is also random. You probably rolled the dice and got lucky the first time
> > 
> > 
> > Yes, but it is definitely consistent with the paintgun held, as far as I've tested it That or I've been unlucky at least ten times in a row
> > However, I think I do recall this crash happening in a casual playthrough in P2:CE, but I reloaded the game and it worked fine so I ended up forgetting about it and I never made an issue about it
> 
> It has nothing to do with holding the paintgun, it's a crash in the physics code. I imagine it's more likely to crash than not. If you don't go through those portals you should be fine

Ok, that makes sense

And yea, I use noclip to get past both of the triggers that trigger the catapault, and it flings me properly

**Comment by JJL772** [comment]
Can you try with `traceray_force_vphysics 0`?

**Comment by JVGRYT** [comment]
> Can you try with `traceray_force_vphysics 0`?

Alright, I gave it a shot. While I had it set to 0, the crash didn't happen at all. But as soon as I turned it back on and re-entered the portal, it crashed again.
